boot/freeldr/freeldr/lib/fs/fat.c

```diff
@@ -1380,8 +1380,8 @@ BOOLEAN FatReadVolumeSectors(PFAT_VOLUME_INFO Volume, ULONG SectorNumber, ULONG
     //
     // Read data
     //
-    Status = ArcRead(Volume->DeviceId, Buffer, SectorCount * 512, &Count);
-    if (Status != ESUCCESS || Count != SectorCount * 512)
+    Status = ArcRead(Volume->DeviceId, Buffer, SectorCount * Volume->BytesPerSector, &Count);
```

Comment:
Oh by the way, are the other supported freeldr FSes use that Volume->BytesPerSector thing too, instead of hardcoding a similar sector size? (By the way I am quite sure ISO fs will hardcode a 2048 size there, I think that value should stay, though.)

Reply:
Yeah, I missed one in FAT (sector count calculation), EXT2 always assumes SECTOR_SIZE and BTRFS uses the SECTOR_SIZE #define everywhere

Reply:
Remember top this code got ripped from xboxdisk.c which had a totally valid reason to assume the sizes.

But UEFI only provides the end result of its test and abstraction of the IO routines. Meaning UEFI will choose whatever the driver for the (UFS in this case) passed it's tests with.

So just using the UEFI provided information is always okay. Because we never access hardware directly.

Reply:
The SECTOR_SIZE here should use device's actual sector size though, using one from boot sector doesn't seem right:
https://github.com/alula/reactos/blob/arm64bringup/boot/freeldr/freeldr/lib/fs/fat.c#L1622

Reply:
This is all stuff we need to consider when we rewrite the disk access code in freeloader. ^